added optional options to noprocess

ricardo118 opened the pull request:
This enables the option &noprocess to be expanded to &noprocess=id,hl

Example:

`[PlayStore](https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de&target=_blank&noprocess)`

Current result link: `https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de&target=_blank` - not opening in new tab


This will work as intended by not processing the `id`, however the target also gets no processed.

Using 

`[PlayStore](https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de&target=_blank&noprocess=id)`

Will skip the `id` processing, and still process the target. Resulting in a better end result link.

New result link: `https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de` - opening in new tab successfully

This is in reference to the following issues:

https://github.com/getgrav/grav/issues/2839 - FIXED

https://github.com/getgrav/grav/issues/2882

https://discordapp.com/channels/501836936584101899/501854266873348112/729728029063905345

ricardo118 commented:
theres some bugs found in discord currently testing

ricardo118 commented:
```
[noprocess](https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de&target=_blank&noprocess)  
result:  `https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de&target=_blank`  
should have `target=_blank` in url and **not open in a new tab**  

[noprocess=id](https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de&target=_blank&noprocess=id)  
result:  `https://play.google.com/store/apps/details?id=org.jitsi.meet&hl=de`  
should have `id=org.jitsi.meet&hl=de` in url and **open in a new tab**  
  
[only target](https://play.google.com/store/apps/details?target=_blank)  
result:  `https://play.google.com/store/apps/details`  
should  **open in a new tab**  
  
[Browser](https://meet.weikamp.biz/Support?rel=nofollow&target=_blank)  
result:  `https://meet.weikamp.biz/Support`  
should  **open in a new tab**  
```

Test cases above

rhukster commented:
be great if you could add these to the test suite @ricardo118

rhukster commented:
Also will need some updates to learn to show how it can be used with options

rhukster commented:
I added some tests, and used those to confirm the functionality is good.  perhaps you can use these for future reference.

tbowan commented:
I needed to remove this "automatic extraction of links attributes" so I removed all of them in the configuration.

Un fortunately, as there is no attributes, the configuration element is no more an array but "null" and I got Exception "in_array() expects parameter 2 to be array, null given" (`/system/src/Grav/Common/Page/Markdown/Excerpts.php` at line 131) :

```php
            // Loop through actions for the image and call them.
            foreach ($actions as $attrib => $value) {
                if (!in_array($attrib, $skip)) {
                    $key = $attrib;
 
                    if (in_array($attrib, $valid_attributes, true)) { // <---- this one is the 131
                        // support both class and classes.
                        if ($attrib === 'classes') {
                            $attrib = 'class';
                        }
                        $excerpt['element']['attributes'][$attrib] = str_replace(',', ' ', $value);
                        unset($actions[$key]);
                    }
                }
            }
```

My workaround without changing the code is to add a random string as a valid attribute in the hope no website will use it as valid attribute (but as I choose it randomly, chances are really tiny).

To fix it in the code, I changed the 117th line as follows (adding the `?: []` at the end) :

```php
            $valid_attributes = $grav['config']->get('system.pages.markdown.valid_link_attributes') ?: [] ;
```

mahagr commented:
@ricardo118 Can you take a look?

ricardo118 commented:
ya will do

ricardo118 commented:
> 
> 
> I needed to remove this "automatic extraction of links attributes" so I removed all of them in the configuration.
> 
> Un fortunately, as there is no attributes, the configuration element is no more an array but "null" and I got Exception "in_array() expects parameter 2 to be array, null given" (`/system/src/Grav/Common/Page/Markdown/Excerpts.php` at line 131) :
> 
> ```
>             // Loop through actions for the image and call them.
>             foreach ($actions as $attrib => $value) {
>                 if (!in_array($attrib, $skip)) {
>                     $key = $attrib;
>  
>                     if (in_array($attrib, $valid_attributes, true)) { // <---- this one is the 131
>                         // support both class and classes.
>                         if ($attrib === 'classes') {
>                             $attrib = 'class';
>                         }
>                         $excerpt['element']['attributes'][$attrib] = str_replace(',', ' ', $value);
>                         unset($actions[$key]);
>                     }
>                 }
>             }
> ```
> 
> My workaround without changing the code is to add a random string as a valid attribute in the hope no website will use it as valid attribute (but as I choose it randomly, chances are really tiny).
> 
> To fix it in the code, I changed the 117th line as follows (adding the `?: []` at the end) :
> 
> ```
>             $valid_attributes = $grav['config']->get('system.pages.markdown.valid_link_attributes') ?: [] ;
> ```
> 
> (I was going to submit a pull request with this change but did not find the Excerps.php file in develop or 1.7 branches so I don't know how to fix it in the github code).

i just tested this, and while yes what you say does happen, i fixed the error without having any issues by simply setting the config to
`valid_link_attributes: []` instead of `valid_link_attributes: null`.

However, admin saves it as null, so i believe the check suggested here is valid.  Pushed a change here [https://github.com/getgrav/grav/commit/2813934d21b04ef5c9a084dfea3e0873588d69e7](https://github.com/getgrav/grav/commit/2813934d21b04ef5c9a084dfea3e0873588d69e7)
